src/hooks/useModal.ts

```diff
@@ -0,0 +1,17 @@
+import { useCallback, useContext } from "react"
+import { Context } from "../providers/ModalsProvider"
+
+const useModal = (
+  modal: React.ReactNode,
+  key?: string,
+): [() => void, () => void] => {
+  const { onClose, onPresent } = useContext(Context)
+
+  const handlePresent = useCallback(() => {
+    onPresent(modal, key)
```

Comment:
What does Present mean? Does that mean like "Show"?

Reply:
Yes, it means "show modal".

Reply:
Does using "show" cause clashing of terms elsewhere? Present technically works by definition but a little strange.